### PR TITLE
Add network tokenization support for auth.net

### PR DESCRIPTION
**Pull request by jnormore**
Docs: http://developer.authorize.net/api/reference/#payment-transactions-charge-a-tokenized-credit-card

@bizla @j-mutter @rwdaigle for review please, thanks!

### CODE REVIEW

File: test/unit/gateways/authorize_net_test.rb [hunk 1]
```diff
@@ -613,6 +613,25 @@ def test_supports_scrubbing?
...
+    credit_card = network_tokenization_credit_card('4242424242424242',
+      :payment_cryptogram => "111111111100cryptogram"
+    )
+
+    response = stub_comms do
+      @gateway.authorize(@amount, credit_card)
+    end.check_request do |endpoint, data, headers|
+      parse(data) do |doc|
+        assert_equal credit_card.payment_cryptogram, doc.at_xpath("//creditCard/cryptogram").content
+        assert_equal credit_card.number, doc.at_xpath("//creditCard/cardNumber").content
```
Comment (j-mutter) [inline test/unit/gateways/authorize_net_test.rb]: This is this expected to be `nil`, yes? Can we maybe just explicitly check that it's blank?

Comment (jnormore) [inline test/unit/gateways/authorize_net_test.rb]: nope, this will be ``4242424242424242``

Comment (j-mutter) [inline test/unit/gateways/authorize_net_test.rb]: Wait, read the docs, this is required when tokenized? Ooooook.

**Comment by j-mutter** [comment]
Rebase + remote test output = :shipit:

**Comment by jnormore** [comment]
```
~/work/active_merchant [apple-pay-authnet]$ be ruby -Itest test/remote/gateways/remote_authorize_net_test.rb -n /test_successful_authorize_and_capture_with_network/
Loaded suite test/remote/gateways/remote_authorize_net_test
Started
.

Finished in 3.044273 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1 tests, 3 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.33 tests/s, 0.99 assertions/s
```

I'm getting failures on the echeck remotes with my credentials though

**Comment by bizla** [comment]
@jnormore for Auth.Net yeah, the blob was ripped from their API docs. 

Test failures aside, code LGTM :+1:

**Comment by rwdaigle** [comment]
@jnormore the docs only show tokenized credit card support for the `authCaptureTransaction` operation, whereas I believe you've implemented it for auth, authCapture and credit (?). Do you know the docs to be incomplete?

Also, I see that for repeat transactions there needs to be a `isPaymentToken` flag set:



We should probably add support for this flag via the options?

**Comment by j-mutter** [comment]
@rwdaigle we can tackle the recurring payments in a future PR - if there is an immediate need I will gladly review one if you put it together :smile:

**Comment by rwdaigle** [comment]
@jnormore completely fair!

**Pull request closed without merging**